File: LAB_9/Adapter.cpp
#include <iostream>
#include <string>

// Interfejs płatności
class PaymentProcessor {
public:
    virtual void processPayment(double amount, std::string currency) = 0;
    virtual ~PaymentProcessor() {}
};

// Istniejący system
class LegacyPaymentProcessor {
public:
    void makePayment(std::string currency, double amount) {
        std::cout << "Processing payment of " << amount << " " << currency << " using legacy system." << std::endl;
    }
};

// Adapter
class PaymentAdapter : public PaymentProcessor {
private:
    LegacyPaymentProcessor* legacyProcessor;
public:
    PaymentAdapter(LegacyPaymentProcessor* legacy) : legacyProcessor(legacy) {}

    void processPayment(double amount, std::string currency) override {
        legacyProcessor->makePayment(currency, amount);
    }
};

// Klient
int main() {
    LegacyPaymentProcessor legacyProcessor;

    PaymentAdapter adapter(&legacyProcessor);

    adapter.processPayment(100, "EUR");

    return 0;
}

File: LAB_9/Adapter.java
// Interfejs płatności
interface PaymentProcessor {
    void processPayment(double amount, String Currency);
}

// Istniejący system
class LegacyPaymentProcessor {
    public void makePayment(String currency, double amount) {
        System.out.println("Processing payment of " + amount + " " + currency + " using legacy system.");
    }
}

// Adapter
class PaymentAdapter implements PaymentProcessor {
    private final LegacyPaymentProcessor legacyProcessor;

    public PaymentAdapter(LegacyPaymentProcessor legacyProcessor) {
        this.legacyProcessor = legacyProcessor;
    }

    @Override
    public void processPayment(double amount, String currency) {
        legacyProcessor.makePayment(currency, amount);
    }
}

// Klient
public class Adapter {
    public static void main(String[] args) {
        LegacyPaymentProcessor legacyProcessor = new LegacyPaymentProcessor();

        PaymentAdapter adapter = new PaymentAdapter(legacyProcessor);

        adapter.processPayment(100, "EUR");
    }
}